controllers/storagecluster/platform_detection.go

```diff
@@ -54,13 +60,64 @@ func (p *Platform) getPlatform(c client.Client) (configv1.PlatformType, error) {
 	return p.platform, nil
 }
 
-func avoidObjectStore(p configv1.PlatformType) bool {
+func avoidObjectStore(p configv1.PlatformType, namespace string, c client.Client) (bool, error) {
```

Comment:
I think we can save ourselves a lot of headache if we make this function a method for `type StorageClusterReconciler`. Both the PlatformType and client.Client come from information in the reconciler. So, instead of taking a PlatformType all you would have to do is pass in a StorageCluster to read the Namespace.

I would request that the refactoring of this function into a method be done as its own commit (without taking a StorageCluster parameter), and then add the IBM changes (with the StorageCluster parameter).

Reply:
@jarrpa Thanks for reviewing the code
Agree that both PlatformType and client.Client come from information in the reconciler. Since we need the platform type in logger, Eg [controllers/storagecluster/cephobjectstores.go]
```
if avoidObjectStore(platform) {
		r.Log.Info(fmt.Sprintf("not creating a CephObjectStore because the platform is '%s'", platform))
		return nil
	}
```
I would return `platform` as well from `avoidObjectStore` method in addition to the bool and error.
In short  fetching the platform part will also be moved to `avoidObjectStore` method so that the code is not repeated in `cephobjectstores.go` file as well as `platform_detection.go` file
```
platform, err := r.platform.GetPlatform(r.Client)
```
Please suggest if this approach is good to go.

Reply:
I don't think this is valid, especially since you just ignore the platform value in multiple places. I think it's fine to duplicate the GetPlatform() line where needed, in this case. and honestly, since you only use the platform for the log message, you can just move the GetPlatform() call to right above the Log.Info() call under the `if` statement.